app/src/lib/markdown-filters/issue-mention-filter.ts

```diff
@@ -0,0 +1,245 @@
+   */
+  private readonly marker = /(?<marker>#|gh-|\/(?:issues|pull|discussions)\/)(?=\d)/
+
+  /**
+   * A regular expression string of a lookbehind is used so that valid matches
+   * for the issue reference have the leader precede them but the leader is not
+   * considered part of the match. An issue reference much have a whitespace,
+   * beginning of line, or some other non-word character must precede it.
+   * */
+  private readonly leader = /(?<=^|\W)/
+
+  /**
+   * A regular expression matching an issue reference.
+   * Issue reference must:
+   * 1) Be preceded by a beginning of a line or some some other non-word
+   *    character.
+   * 2) Start with an issue marker: gh-, #, /issues/, /pull/, or /discussions/
+   * 3) The issue marker must be followed by a number
+   * 4) The number must end in a word bounding character. Additionally, the
+   *    issue reference match may be such that the marker may be preceded by a
+   *    repo references of owner/repo or owner
+   * */
+  private readonly issueReferenceWithLeader = new RegExp(
+    this.leader.source +
+      this.ownerOrOwnerRepo.source +
+      '?' +
+      this.marker.source +
+      this.refNumber.source,
+    'ig'
+  )
+
+  /** The parent github repository of which the content the filter is being
+   * applied to belongs  */
+  private readonly repository: GitHubRepository
+
+  public constructor(repository: GitHubRepository) {
+    this.repository = repository
+  }
+
+  /**
+   * Returns tree walker that iterates on all text nodes that are not inside a
+   * pre, code, or anchor tag.
+   */
+  public createFilterTreeWalker(doc: Document): TreeWalker {
+    return doc.createTreeWalker(doc, NodeFilter.SHOW_TEXT, {
+      acceptNode: function (node) {
+        return node.parentNode !== null &&
+          ['CODE', 'PRE', 'A'].includes(node.parentNode.nodeName)
+          ? NodeFilter.FILTER_SKIP
+          : NodeFilter.FILTER_ACCEPT
+      },
+    })
+  }
+
+  /**
+   * Takes a text node and creates multiple text and anchor nodes by inserting
+   * anchor tags where the matched issue mentions exist.
+   *
+   * Warning: This filter can create false positives. It assumes the issues
+   * exists that are mentioned. Thus, if a user references a non-existent
+   * #99999999 issue, it will still create a link for it. This is a deviation
+   * from dotcoms approach that verifies each link, but we do not want to incur
+   * the performance penalty of making that call.
+   *
+   * Example:
+   * Node = "Issue #1234 is the same thing"
+   * Output = ["Issue ", <a href="https://github.com/owner/repo/issues/1234">#1234</a>, " is the same thing"]
+   */
+  public async filter(node: Node): Promise<ReadonlyArray<Node> | null> {
+    const { textContent: text } = node
+    const markerRegexp = new RegExp(this.marker, 'i')
+    if (
+      node.nodeType !== node.TEXT_NODE ||
+      text === null ||
+      !markerRegexp.test(text)
+    ) {
+      return null
+    }
```

Comment:
Maybe it's worth doing the `nodeType` and `text` checks before creating the `RegExp` object, since it's expensive to create. It'd be even better if we could cache it.
```suggestion
    if (
      node.nodeType !== node.TEXT_NODE ||
      text === null
    ) {
      return null
    }
    const markerRegexp = new RegExp(this.marker, 'i')
    if (!markerRegexp.test(text)) {
      return null
    }
```

Reply:
I refactored slightly to make it so we just use the `marker` regex as defined on the filter instantiation. That way with the node filter being cached. The regex should only be created once.